lib/logproto/logproto-client.h

```diff
@@ -112,13 +112,18 @@ log_proto_client_validate_options(LogProtoClient *self)
   return self->validate_options(self);
 }
 
+static inline gboolean
+log_proto_client_needs_handshake(LogProtoClient *s)
+{
+  return s->handshake != NULL;
+}
+
 static inline LogProtoStatus
 log_proto_client_handshake(LogProtoClient *s, gboolean *handshake_finished)
 {
-  if (s->handshake)
-    {
-      return s->handshake(s, handshake_finished);
-    }
+  if (log_proto_client_needs_handshake(s))
```

Comment:
all our method wrappers check the methods directly and not via another helper function. are there any other uses for log_proto_client_needs_handshake()?

Reply:
- it is an inline func, no penalities at all
- it can be a more complex condition for a given proto client, and this can be an overriden impl later if needed
- yes, it is reused and called from other places